src/components/approval-tasks/ApprovalTaskActionDropdown.tsx

```diff
@@ -84,7 +112,10 @@ const ApprovalTaskActionDropdown: React.FC<ApprovalTaskActionDropdownProps> = ({
     if (state !== ApproverStatusResponse.Pending) {
       return t(`PipelineRun has been {{state}}`, { state });
     }
-    if (!approvers?.find((approver) => approver === currentUser)) {
+    if (isAuthorized === null) {
+      return t('Checking authorization...');
```

Comment:
For better UX we can add a [Spinner](https://patternfly.org/components/spinner) or something here